### PR TITLE
tiny: add module

### PR DESCRIPTION
**Pull request by kmaasrud**
### Description

Added module for [tiny](https://github.com/osa1/tiny), a TUI IRC client written in Rust. Thanks to @ratsclub for helping me write my first Nix module!

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Review by otavio — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -437,3 +437,5 @@ Makefile                                              @thiagokokada
 /modules/programs/ion.nix                             @jo1gi
 
 /modules/services/plex-mpv-shim.nix                   @starcraft66
+
+/modules/programs/tiny.nix                            @kmaasrud
```
Comment (berbiche) [inline .github/CODEOWNERS:441]: Could you move this line after `/modules/programs/timidity.nix @amesgen` to preserve the ordering as much as possible?

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the contribution!

The module follows all our standards :+1:
The only nit is that the line added to the CODEOWNERS file is not at the right place.

File: .github/CODEOWNERS (hunk 1 above)
Comment (kmaasrud) [reply .github/CODEOWNERS]: Ofc! This should now be fixed 😄

**Comment by kmaasrud** [comment]
I also added the darwin configuration path fix from #2736. tiny seems to respect [standard directories](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW6), making XDG useless on Darwin.

**Review by ratsclub — APPROVED** [review]

File: modules/programs/tiny.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      package = mkOption {
+        type = types.package;
+        default = pkgs.tiny;
+        defaultText = literalExpression "pkgs.tiny";
+        description = "The tiny package to install.";
+      };
+
+      settings = mkOption {
+        type = format.type;
+        default = { };
```
Comment (sumnerevans) [inline modules/programs/tiny.nix:27]:
```suggestion
        default = { };
        defaultText = literalExpression "{ }";
```

File: modules/programs/tiny.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  configDir = if pkgs.stdenv.isDarwin then
+    "Library/Application Support/tiny"
+  else
+    "${config.xdg.configHome}/tiny";
+in {
+  meta.maintaners = [ maintainers.kmaasrud ];
+
+  options = {
+    programs.tiny = {
+      enable = mkEnableOption "tiny";
```
Comment (sumnerevans) [inline modules/programs/tiny.nix:16]:
```suggestion
      enable = mkEnableOption "tiny, a TUI IRC client written in Rust";
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

**Comment by sumnerevans** [comment]
Looks like you need to add yourself as a maintainer. Note that I just rebased your branch and force-pushed due to unrelated build failures that just got fixed.

**Comment by kmaasrud** [comment]
Thanks for that @sumnerevans! Added myself to `maintainers.nix` now

File: modules/programs/tiny.nix [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+with lib;
+let
+  cfg = config.programs.tiny;
+  format = pkgs.formats.yaml { };
+  configDir = if pkgs.stdenv.isDarwin then
+    "Library/Application Support/tiny"
+  else
+    "${config.xdg.configHome}/tiny";
+in {
+  meta.maintainers = [ maintainers.kmaasrud ];
```
Comment (sumnerevans) [inline modules/programs/tiny.nix:12]: I think you need 
```suggestion
  meta.maintainers = [ hm.maintainers.kmaasrud ];
```

File: modules/programs/tiny.nix [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        type = types.package;
+        default = pkgs.tiny;
+        defaultText = literalExpression "pkgs.tiny";
+        description = "The <command>tiny</command> package to install.";
+      };
+
+      settings = mkOption {
+        type = format.type;
+        default = { };
+        defaultText = literalExpression "{  }";
```
Comment (sumnerevans) [inline modules/programs/tiny.nix:28]:
```suggestion
        defaultText = literalExpression "{ }";
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

**Comment by kmaasrud** [comment]
Sorry about all the fuzz; I'm a new contributor, as you understand 😅

**Review by sumnerevans — APPROVED** [review]
LGTM

**Pull request merged**